scripts/check_lint.py

```diff
@@ -0,0 +1,232 @@
+
+    # cpplint is generally confused by Objective-C mixing with C++.
+    #   * Objective-C method invocations in a for loop make it think its a
+    #     range-for
+    #   * Objective-C dictionary literals confuse brace spacing
+    #   * Empty category declarations ("@interface Foo ()") look like function
+    #     invocations
+    '-whitespace',
+]
+
+_CPPLINT_OBJC_OPTIONS = [
+    # cpplint normally excludes Objective-C++
+    '--extensions=h,m,mm',
+
+    # Objective-C style allows longer lines
+    '--linelength=100',
+
+    '--filter=' + ','.join(_CPPLINT_OBJC_FILTERS),
+]
+
+
+def main():
+  global _dry_run
+
+  parser = argparse.ArgumentParser(description='Lint source files.')
+  parser.add_argument('--dry-run', '-n', action='store_true',
+                      help='Show what the linter would do without doing it')
+  parser.add_argument('--all', action='store_true',
+                      help='run the linter over all known sources')
+  parser.add_argument('rev_or_files', nargs='*',
+                      help='A single revision that specifies a point in time '
+                           'from which to look for changes. Defaults to '
+                           'origin/master. Alternatively, a list of specific '
+                           'files or git pathspecs to lint.')
+  args = command_trace.parse_args(parser)
+
+  if args.dry_run:
+    _dry_run = True
+    command_trace.enable_tracing()
+
+  pool = checker.Pool()
+
+  sources = _unique(source.CC_DIRS + source.OBJC_DIRS + source.PYTHON_DIRS)
+  patterns = git.make_patterns(sources)
+
+  files = git.find_changed_or_files(args.all, args.rev_or_files, patterns)
+  check(pool, files)
+
+  pool.exit()
+
+
+def check(pool, files):
+  group = source.categorize_files(files)
+
+  for kind, files in group.kinds.items():
+    for chunk in checker.shard(files):
+      if not chunk:
+        continue
+
+      linter = _linters[kind]
+      pool.submit(linter, chunk)
+
+
+def lint_cc(files):
+  return _run_cpplint([], files)
+
+
+def lint_objc(files):
+  return _run_cpplint(_CPPLINT_OBJC_OPTIONS, files)
+
+
+def _run_cpplint(options, files):
+  scripts_dir = os.path.dirname(os.path.abspath(__file__))
+  cpplint = os.path.join(scripts_dir, 'cpplint.py')
+
+  command = [sys.executable, cpplint, '--quiet']
+  command.extend(options)
+  command.extend(files)
+
+  return _read_output(command)
+
+
+_flake8_warned = False
+
+
+def lint_py(files):
+  flake8 = which('flake8')
+  if flake8 is None:
+    global _flake8_warned
+    if not _flake8_warned:
+      _flake8_warned = True
+      _logger.warn(textwrap.dedent(
+          """
+          Could not find flake8 on the path; skipping python lint.
+          Install with:
+
+            pip install --user flake8
+          """))
+    return
+
+  command = [sys.executable, flake8]
+  command.extend(files)
+
+  return _read_output(command)
+
+
+def _read_output(command):
+  command_trace.log(command)
+
+  if _dry_run:
+    return 0
```

Comment:
```
$ ./scripts/check_lint.py -n
<cut>
Traceback (most recent call last):
<cut>
  File "scripts/lib/checker.py", line 109, in join
    num_errors += result.errors
AttributeError: 'int' object has no attribute 'errors'
```

The problem obviously being that you're returning an int here, rather than a checker.Result.

Reply:
Fixed.